app/src/full/java/io/homeassistant/companion/android/settings/wear/views/SettingsWearHomeView.kt

```diff
@@ -48,30 +50,51 @@ fun LoadSettingsHomeView(
                     onBackClicked = onStartBackClicked,
                     events = settingsWearViewModel.resultSnackbar
                 )
             }
-            composable(SettingsWearMainView.TEMPLATE) {
-                SettingsWearTemplateTile(
-                    template = settingsWearViewModel.templateTileContent.value,
-                    renderedTemplate = settingsWearViewModel.templateTileContentRendered.value,
-                    refreshInterval = settingsWearViewModel.templateTileRefreshInterval.value,
-                    onContentChanged = {
-                        settingsWearViewModel.setTemplateContent(it)
-                        settingsWearViewModel.sendTemplateTileInfo()
-                    },
-                    onRefreshIntervalChanged = {
-                        settingsWearViewModel.templateTileRefreshInterval.value = it
-                        settingsWearViewModel.sendTemplateTileInfo()
+            composable(SettingsWearMainView.TEMPLATES) {
+                SettingsWearTemplateTileList(
+                    templateTiles = settingsWearViewModel.templateTiles,
+                    onTemplateTileClicked = { tileId ->
+                        navController.navigate(SettingsWearMainView.TEMPLATE_TILE.format(tileId))
                     },
                     onBackClicked = {
                         navController.navigateUp()
                     }
                 )
             }
+            composable(
+                route = SettingsWearMainView.TEMPLATE_TILE.format("{tileId}"),
+                arguments = listOf(navArgument("tileId") { type = NavType.IntType })
+            ) { backStackEntry ->
+                val tileId = backStackEntry.arguments?.getInt("tileId")
+                val templateTile = settingsWearViewModel.templateTiles[tileId]
+                val renderedTemplate = settingsWearViewModel.templateTilesRenderedTemplates[tileId]
+
+                // TODO: error handling
+                templateTile?.let {
+                    SettingsWearTemplateTile(
+                        template = it.template,
+                        renderedTemplate = renderedTemplate ?: "",
+                        refreshInterval = it.refreshInterval,
+                        onContentChanged = { templateContent ->
+                            settingsWearViewModel.setTemplateTileContent(tileId!!, templateContent)
+                            settingsWearViewModel.sendTemplateTileInfo()
+                        },
+                        onRefreshIntervalChanged = { refreshInterval ->
+                            settingsWearViewModel.setTemplateTileRefreshInterval(tileId!!, refreshInterval)
+                            settingsWearViewModel.sendTemplateTileInfo()
+                        },
+                        onBackClicked = {
+                            navController.navigateUp()
+                        }
+                    )
+                }
```

Comment:
Any suggestions on how to handle errors here? I guess it's more of a theoretical problem now that no stale tile entries should be left in the app state.

Reply:
I am not sure if error handling is needed here as you already have a null check. Have you tested the behavior if a user  had 1 template tile before the upgrade and looked at phone settings before opening the app on the watch? I would imagine if there is any error it would possibly be from existing data